access-protected-api/README.md

```diff
@@ -0,0 +1,162 @@
+
+### 2. Update code sample with app registration values
+
+Open the [_application.yml_](src/main/resources/application.yml) file and modify the four Azure Active Directory configuration properties using the values from your [app's registration in the Azure portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-configure-app-expose-web-apis).
+
+```yaml
+# 'Tenant ID' of your Azure AD instance - this value is a GUID
+tenant-id: value-here
+
+# 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+client-id: value-here
+
+# Client secret 'Value' (not its ID) from 'Client secrets' in app registration in Azure portal
+client-secret: value-here
+
+# 'Object ID' of app registration in Azure portal - this value is a GUID
+client-object-id: value-here
+```
+
+### 3. Install package(s) and compile application
+
+To install Spring Boot and MSAL libraries:
+
+```bash
+mvn install
+```
+
+## Run the application
+
+```bash
+mvn spring-boot:run
+```
+
+## Access the API
+
+Using Postman, curl, or a similar application, issue an HTTP GET request to *http://localhost:8080/api/application*.
+
+For example, if you use curl and everything worked, the sample you should receive a response from the API similar to this:
+
+```console
+$ curl http://localhost:8080/api/application
+{
+  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#applications/$entity",
+  "id": "04487cab-91cf-46d8-b4ed-1a428388bdd2",
+  "appId": "de5dcdc6-8dfe-4048-aff6-c615dea4ee3d",
+  "displayName": "java-api",
+  "description": null,
+  "groupMembershipClaims": null,
+  "identifierUris": ["api://de5dcdc6-8dfe-4048-aff6-c615dea4ee3d"],
+  "isDeviceOnlyAuthSupported": null,
+  "isFallbackPublicClient": null,
+  "notes": null,
+  "publisherDomain": "contoso.onmicrosoft.com",
+  "signInAudience": "AzureADMyOrg",
+  "tags": [],
+  "tokenEncryptionKeyId": null,
+  "defaultRedirectUri": null,
+  "certification": null,
+  "optionalClaims": null,
+  …
+}
+```
+
+## About the code
+
+This Java API uses the Spring Boot web framework. The app has a single, anonymous route. When that route is called, the application will make a Microsoft Graph request to get information about its own app registration. This is being done via the client credentials OAuth flow, meaning the application is using its own App registration as the credentials to generate the access token for Microsoft Graph. The results of the Microsoft Graph call are then returned as the results of the API call.
+
+No specific Microsoft Graph permissions needed to be assigned to the app registration in the Azure AD portal, because app registrations can access their own records without additional permissions. To access other Microsoft Graph endpoints, the app registration for this API would require additional Microsoft Graph API permissions assigned to it.

```

Comment:
```suggestion
The web API requires no specific Microsoft Graph API permissions because applications registered with Azure Azure AD always have permission to access their own record, or *object*, in Microsoft Graph. To access other Microsoft Graph API endpoints, however, permissions for those endpoints must be added to the web API's registration in Azure AD.
```

Reply:
Nice.